plugins/npm/nexus-npm-repository-plugin/src/main/java/com/bolyuba/nexus/plugin/npm/service/internal/orient/OrientMetadataStore.java

```diff
@@ -113,6 +128,46 @@ public void doStop() throws Exception {
   }
 
   /**
+   * Performs DB backup, during which database is "frozen" for data modification attempts.
+   *
+   * @since 2.11
+   */
+  public void backupDatabase() throws IOException {
+    try (final ODatabaseDocumentTx db = db()) {
+      final File backupFile = createBackupFile();
+      log.info("Started npm DB backup to {} (all DB write operations are frozen)",
+          backupFile.getAbsolutePath());
+      final OutputStream outputStream = new FileOutputStream(backupFile);
```

Comment:
Yes, db.backup wraps the stream (for buffering reasons), and closes it on return.